pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -727,12 +730,14 @@ func DeleteXfrm() {
 		}
 	}
 	xfrmStateList, err := netlink.XfrmStateList(netlink.FAMILY_ALL)
-	if err == nil {
-		for _, s := range xfrmStateList {
-			if isXfrmStateCilium(s) {
-				if err := netlink.XfrmStateDel(&s); err != nil {
-					log.WithError(err).Warning("deleting old xfrm state failed")
-				}
+	if err != nil {
+		log.WithError(err).Warning("unable to fetch xfrm state list")
+		return
+	}
+	for _, s := range xfrmStateList {
+		if isXfrmStateCilium(s) {
+			if err := netlink.XfrmStateDel(&s); err != nil {
+				log.WithError(err).Warning("deleting old xfrm state failed")
```

Comment:
This is a good step, but I'd go further: let's make this function return an error and return an aggregated error of how many resources failed to be deleted and a record of the error types observed similar to my above comment. Then we can move the logging of the error to the caller of this function, rather than on each failure to delete an individual resource. I think this makes sense given the role of this function as a "leaf" in terms of XFRM logic and interaction with netlink.

Reply:
Hum... I was not sure about this part. The original code just logged and did not propagate. I am guessing because we want to add scope to the logs which will be lost unless we wrap the error further?? But would kill the log indexer is that is the intent here??

Reply:
Yeah, my suggestion would be to try to prevent as much of this pattern where the leaf functions are logging and instead kick the error back up to the caller. Then the caller can decide to log or not. There can be exceptions to this rule depending on the leaf function, but in this case, it seems like it could do with the same thing we implemented in `xfrmDeleteConflictingState()`

Reply:
Makes sense. I've changes the call to now return an error.